pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConnectionHandler.java

```diff
@@ -63,11 +64,20 @@ protected void grabCnx() {
             return;
         }
 
-        if (!isValidStateForReconnection()) {
-            // Ignore connection closed when we are shutting down
-            log.info("[{}] [{}] Ignoring reconnection request (state: {})",
-                    state.topic, state.getHandlerName(), state.getState());
-            return;
+        synchronized (this) {
+            if (connectFuture != null && !connectFuture.isDone()) {
+                log.warn("[{}] [{}] Skip grabbing the connection since there is a pending connection",
```

Comment:
I guess this is a normal case that we don't need warning?

Reply:
Yes.